src/imgui/ImGuiSymbols.cc

```diff
@@ -188,16 +268,58 @@ void ImGuiSymbols::paint(MSXMotherBoard* /*motherBoard*/)
 		}
 
 		im::TreeNode("Symbols per file", ImGuiTreeNodeFlags_DefaultOpen, [&]{
-			auto drawFile = [&](const FileInfo& info) {
+			auto drawFile = [&](FileInfo& info) {
+				auto fileIdx = symbolManager.findFile(info.filename);
 				im::StyleColor(!info.error.empty(), ImGuiCol_Text, getColor(imColor::ERROR), [&]{
 					auto title = strCat("File: ", info.filename);
 					im::TreeNode(title.c_str(), [&]{
 						if (!info.error.empty()) {
 							ImGui::TextUnformatted(info.error);
 						}
 						im::StyleColor(ImGuiCol_Text, getColor(imColor::TEXT), [&]{
+							auto& cpuInterface = motherBoard->getCPUInterface();
+							std::string slotInfo = {'-', '\000'};
+							std::vector<std::pair<std::optional<uint16_t>, std::optional<uint16_t>>> slotSubslots;
+							slotSubslots.push_back({{}, {}}); // initial state
+							int nSlots = 1;
+
+							for (uint16_t ps = 0; ps < 4; ++ps) {
+								if (cpuInterface.isExpanded(ps)) {
+									slotInfo = tmpStrCat(slotInfo, ps, "-0", '\000', ps, "-1", '\000', ps, "-2", '\000', ps, "-3", '\000');
+									slotSubslots.push_back({ps, 0});
+									slotSubslots.push_back({ps, 1});
+									slotSubslots.push_back({ps, 2});
+									slotSubslots.push_back({ps, 3});
+									nSlots += 4;
+								} else {
+									slotInfo = tmpStrCat(slotInfo, ps, '\000');
+									slotSubslots.push_back({ps, {}});
+									nSlots++;
+								}
+							}
+							auto arrowSize = ImGui::GetFrameHeight();
+							auto extra = arrowSize + 2.0f * style.FramePadding.x;
+							ImGui::SetNextItemWidth(ImGui::CalcTextSize("3-3").x + extra);
+						        ImGui::AlignTextToFramePadding();
+
+							// slot
+							ImGui::Combo(tmpStrCat("Slot##", info.filename).data(), &info.index, slotInfo.c_str(), nSlots);
+							if (fileIdx.has_value()) {
+								auto& file = symbolManager.getFiles()[*fileIdx];
+								// detect slot/subslot changes
+								if (file.slot != slotSubslots[info.index].first || file.subslot != slotSubslots[info.index].second) {
+									file.slot = slotSubslots[info.index].first;
+									file.subslot = slotSubslots[info.index].second;
+									for (auto& symbol: file.getSymbols()) {
+										symbol.slot = slotSubslots[info.index].first;
+										symbol.subslot = slotSubslots[info.index].second;
+									}
+								}
```

Comment:
I mean this stuff. You only need to do this when the slot combo box was actually changed. And that's what the `bool` result from `ImGui::Selectable()` tells you.